### DISCUSSION TITLE
Adding a substantially similar / forked linter to the project

### DISCUSSION THREAD

**Discussion by ImprintNav** [golangci/golangci-lint#4076] [Q&A] [answered]
I'm thinking of adding a linter that is substantially similar to an existing linter because the maintainer of the original is not willing to add or willing to respond to requests to add certain features that would be crucial to a variety of projects. Is this a faux pas?

**Comment by ldez** [comment]
Hello,

can you give me the name of this linter?

We won't accept a fork just like that, we need to communicate beforehand.

In all cases, if a fork is needed, we will create this fork inside the organization.

Also, we don't allow duplicate linter.

**Comment by ImprintNav** [comment]
This would be exhaustruct.

**Reply by ldez** [reply]
Can you be more explicit: What is your problem?
is it [this](https://github.com/GaijinEntertainment/go-exhaustruct/issues/66) issue?

You have done 0 PR on this repo and the maintainer has answered your issue, so we will not accept a fork because you can open a PR on this repo.

If you have doubts, I recommend opening a dedicated issue on the [exhaustruct](https://github.com/GaijinEntertainment/go-exhaustruct) repository to ask if the repo is still maintained.

**Reply by bombsimon** [reply]
Interesting. `exhaustruct` is a linter that got added in #2667 because `exhaustivestruct` was unmaintained. Would be a real shame to have to do this again just a year and a half later. It looks like https://github.com/GaijinEntertainment/go-exhaustruct/ both have issues with replies and commits pushed to `master` in end of July, that's just 1,5 months ago.

What issues/PR are you referring to?

**Reply by ImprintNav** [reply]
See GaijinEntertainment/go-exhaustruct#37, GaijinEntertainment/go-exhaustruct#69, GaijinEntertainment/go-exhaustruct#70. All these issues and PRs are under my accounts.

This might be just a simple disagreement on the project direction but it's a shame that the author is seemingly not open to discussions beyond bug fixes. This is something I have wanted to contribute since nishanths/exhaustive#40, but unfortunately the idea has already been claimed and incorporated into golangci-lint under arguably not the most open source culture.

This was probably a longshot to ask for anyway. Though I would implore considering the opportunity cost incurred by users in having this analyzer pigeonholed to a limited feature set and a fairly closed development cycle.

**Reply by ImprintNav** [reply]
Update: case in point, the repository doesn't seem to support creating pull requests anymore if you aren't a collaborator. As far as I know there are no collaborators other than the owner.

**Reply by ldez** [reply]
> Update: case in point, the repository doesn't seem to support creating pull requests anymore if you aren't a collaborator. As far as I know there are no collaborators other than the owner.

It's always possible to create a PR ([I did it](https://github.com/GaijinEntertainment/go-exhaustruct/pull/76) and I have no rights on this repo), maybe you have been banned from this repo.

> This might be just a simple disagreement on the project direction but it's a shame that the author is seemingly not open to discussions beyond bug fixes. 

Or maybe not.
Maybe it's related to your behavior, maybe it's because the author doesn't have time, maybe it's because the author doesn't have to fix everything, etc.
There are no requirements to communicate, to discuss, to explain, or to agree.
https://github.com/readme/featured/how-open-is-open-source
The maintenance of OSS is complex and there is no "shame" because a project doesn't follow your ideas or your way of seeing OSS.

Your hypothesis seems wrong, so your conclusion is also wrong.

---

The usage of multiple GitHub accounts to interact with one repo can be seen as a form of harassment.

The problem is neither the linter nor the maintenance of this linter.

I recommend thinking about your behavior and trying to talk with the author of exhaustruct.

**Reply by ImprintNav** [reply]
> The usage of multiple GitHub accounts to interact with one repo can be seen as a form of harassment

I have not presented myself as multiple people at any point. I am on my work account at some points and on my personal accounts at others. My username is not intended to disguise my identity. Is there any guideline published that you should not interact with a repo using multiple accounts? If so I am happy to only stick to my personal account in the future.

> I recommend thinking about your behavior and trying to talk with the author of exhaustruct.

What is there to think about? I created issues first and PRs some time subsequently after time for the owner to respond. There was no pushback against any of these, aside from one PR that the author closed and replaced with his own to address the underlying issue (which I also raised). I have done the same for related analyzers e.g. nishanths/exhaustive#36; there was no difference in my behavior, only in the repo owner's.

On your other points, this is a valid point of disagreement. It seems this is consistent with the OSS culture of the golangci-lint project as a whole. I will simply have to create my own fork of the repo and incorporate it manually into golangci-lint in future projects. Thanks for your time discussing this issue.

**Reply by ldez** [reply]
> I have not presented myself as multiple people at any point. 

This is the problem: You are using different accounts to interact on an issue which creates fake traction on the topic.
This is the behavior of someone who wants to put pressure on the author. I don't say that you intentionally did that but you did it.
I recommend reading documents or watching videos about harassment.

> My username is not intended to disguise my identity.

But you did it. No one can know that you are behind those different usernames.

> On your other points, this is a valid point of disagreement.

I think you didn't read the article and the stories of those OSS maintainers.
The world is not in 2 colors.
And to be clear, I just tried to explain.
Everybody is different and we have to respect that.

> It seems this is consistent with the OSS culture of the golangci-lint project as a whole.

Can you clarify this sentence?

>  I will simply have to create my own fork of the repo and incorporate it manually into golangci-lint in future projects. 

I think you don't understand my suggestions.
I will do the job for you, and try to communicate with the author of exhaustruct.

EDIT: https://github.com/GaijinEntertainment/go-exhaustruct/discussions/77

**Reply by navijation** [reply]
> But you did it. No one can know that you are behind those different usernames.

See GaijinEntertainment/go-exhaustruct#63. The commits are under my work account. My first name prefix is the same as my work account suffix. I always respond as a single person throughout these conversations and. am not liking my own comments or PRs / issues. I've done this before e.g. bazelbuild/rules_go#3082 with no ensuing confusion. However, I'll refrain in the future to avoid ambiguity.

> I think you don't understand my suggestions.
I will do the job for you, and try to communicate with the author of exhaustruct.

I sent an email to the author regarding this issue, Tweets to the author in the past, and engaged in the Github issues. I'm pretty sure that is what you are suggesting. I'll forward you my email from earlier today.

> Everybody is different and we have to respect that.

I said I respect _your_ disagreement and believe it's valid. We're on the same page here.

> Can you clarify this sentence?

I mean that the author's maintenance of the analyzer is not inconsistent with the expectations of this project as a whole.

**Reply by ldez** [reply, marked as answer]
The answer of the author:

> I simply had no spare time since the middle of july and just returned from long-awaited vacation🙃
>
> About user that claims to be banned - i didnt do anything proactively to ban him, interaction limits are off and i cant see any other config that can prevent exact user from creating PRs🤔

https://github.com/GaijinEntertainment/go-exhaustruct/discussions/77#discussioncomment-6964157
